packages/@aws-cdk/aws-codestarnotifications/lib/target.ts

```diff
@@ -0,0 +1,28 @@
+import { INotificationRule } from './notification-rule';
+
+/**
+ * Information about the SNS topic or AWS Chatbot client associated with a notification target.
+ */
+export interface NotificationRuleTargetConfig {
+  /**
+   * The target type. Can be an Amazon SNS topic or AWS Chatbot client.
+   */
+  readonly targetType: string;
+
+  /**
+   * The Amazon Resource Name (ARN) of the Amazon SNS topic or AWS Chatbot client.
+   */
+  readonly targetAddress: string;
+}
+
+/**
+ * Represents a notification target
+ * That allows AWS Chatbot and SNS topic to associate with this rule target.
+ */
+export interface INotificationRuleTarget {
+  /**
+   * Binds target to notification rule
+   * @param _rule The notification rule
+   */
+  bind(_rule: INotificationRule): NotificationRuleTargetConfig;
```

Comment:
1. This should be called something more distinctive, so that it doesn't clash with other `bind()` methods. Something like `bindAsNotificationRuleTarget()`.
2. Make it also take a `scope: Construct` as the first argument.

Reply:
I like this naming `bindAs*` , it can make more flexible to bind other modules.